recipe/meta.yaml

```
@@ -1,57 +1,100 @@
-{% set name = "msgspec" %}
+{% set name = "msgspec-packages" %}
 {% set version = "0.18.6" %}
+{% set name_base = "msgspec" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: a59fc3b4fcdb972d09138cb516dbde600c99d07c38fd9372a6ef500d2d031b4e
+  # Use GH to include upstream tests
+  url: https://github.com/jcrist/{{ name_base }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 5bff8c3f9303eb589516e02de4141ce31be09a0fdf4980734b020b64c120295e
 
 build:
   number: 0
+  # Skip s390x as it's optional and fails some tests
+  skip: true  # [py<38 or s390x]
+requirements:
+  host:
+    - python
 
 outputs:
-  - name: msgspec
+  - name: {{ name_base }}
     version: {{ version }}
     build:
-      skip: true  # [py<38]
-      script: python -m pip install . -vv
+      script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
     requirements:
       build:
-        - python                                 # [build_platform != target_platform]
-        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
         - {{ compiler('c') }}
       host:
         - python
         - pip
+        - setuptools
+        - wheel
       run:
         - python
     test:
       imports:
         - msgspec
+      requires:
+        - pip
+        - pytest
+      source_files:
+        - tests
+        - msgspec/_core.c
+        - setup.cfg
+      commands:
+        - pip check
+        - pytest tests
 
   - name: msgspec-yaml
```

```suggestion
  - name: {{ name }}-yaml
```

just for consistency, nothing major